expression/integration_test.go

```diff
@@ -1396,6 +1396,41 @@ func (s *testIntegrationSuite) TestBuiltin(c *C) {
 	result = tk.MustQuery("select cast('-34 100:00:00' as time);")
 	result.Check(testkit.Rows("-838:59:59"))
 
+	// fix issue #4324. cast decimal/int/string to time compability.
+	invalidTimes := []string{
+		"10009010",
+		"239010",
+		"233070",
+		"23:90:10",
+		"23:30:70",
+		"239010.2",
+		"233070.8",
+	}
+	tk.MustExec("DROP TABLE IF EXISTS t;")
+	tk.MustExec("CREATE TABLE t (ix TIME);")
+	tk.MustExec("SET SQL_MODE='';")
+	for _, invalidTime := range invalidTimes {
+		msg := fmt.Sprintf("Warning 1292 Truncated incorrect time value: '%s'", invalidTime)
+		result = tk.MustQuery(fmt.Sprintf("select cast('%s' as time);", invalidTime))
+		result.Check(testkit.Rows("00:00:00"))
```

Comment:
in mysql:
```sql
MySQL > select cast('10009010' as time);
+--------------------------+
| cast('10009010' as time) |
+--------------------------+
| NULL                     |
+--------------------------+
1 row in set, 1 warning (0.00 sec)

MySQL > set sql_mode='';
Query OK, 0 rows affected, 1 warning (0.01 sec)

MySQL > select cast('10009010' as time);
+--------------------------+
| cast('10009010' as time) |
+--------------------------+
| NULL                     |
+--------------------------+
1 row in set, 1 warning (0.00 sec)
```

the test case is not compatible with mysql's behavior

Reply:
@zz-jason PTAL. 
changes:
1.when return value is ZeroDuration, then set isNull to true.
2.change the testcase.